homeassistant/components/climate/spider.py

```diff
@@ -0,0 +1,151 @@
+    SUPPORT_TARGET_TEMPERATURE, ClimateDevice)
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME, TEMP_CELSIUS
+import homeassistant.helpers.config_validation as cv
+from homeassistant.util import Throttle
+
+REQUIREMENTS = ['itho_daalderop_api==1.0.3']
+
+_LOGGER = logging.getLogger(__name__)
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(minutes=15)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Spider thermostat."""
+    from itho_daalderop_api import IthoDaalderop_API, UnauthorizedException
+
+    username = config.get(CONF_USERNAME)
+    password = config.get(CONF_PASSWORD)
+
+    try:
+        api = IthoDaalderop_API(username, password)
+
+    except UnauthorizedException:
+        _LOGGER.exception("Can't connect to the Itho Daalderop API")
+        return
+
+    thermostats = api.get_thermostats()
+    for i, thermostat in enumerate(thermostats):
+        add_devices(
+            [SpiderThermostat(api, thermostat['id'],
+                              thermostat['name'], i == 0)],
+            True
+        )
+
+
+class SpiderThermostat(ClimateDevice):
+    """Representation of a thermostat."""
+
+    def __init__(self, client, thermostat_id, name, master):
+        """Initialize the thermostat."""
+        self.client = client
+        self._id = thermostat_id
+        self._name = name
+        self._master = master
+        self._thermostat = None
+        self._current_temperature = None
+        self._target_temperature = None
+        self._min_temp = None
+        self._max_temp = None
+        self._operation = STATE_IDLE
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        supported = (SUPPORT_TARGET_TEMPERATURE)
+
+        return supported
+
+    @property
+    def name(self):
+        """Return the name of the thermostat, if any."""
+        return self._name
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement."""
+        return TEMP_CELSIUS
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return self._current_temperature
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        return self._target_temperature
+
+    @property
+    def min_temp(self):
+        """Return the minimum temperature."""
+        return self._min_temp
+
+    @property
+    def max_temp(self):
+        """Return the maximum temperature."""
+        return self._max_temp
+
+    def set_temperature(self, **kwargs):
+        """Set new target temperature."""
+        temperature = kwargs.get(ATTR_TEMPERATURE)
+        if temperature is None:
+            return
+
+        self._target_temperature = temperature
```

Comment:
It's better to let `update` update the instance attributes. `update` will be called immediately after each actuator call during service calls.

Reply:
done